python/cairo-core/src/cairo_core/control_flow.cairo

```diff
@@ -0,0 +1,19 @@
+// @dev Raise by asserting [ap - 1] == [ap - 2];
+//      The instruction is dw to have the last instruction run failing,
+//      while using assert [ap - 1] == [ap - 2] would require to add ret;
+//      at the end of the function, which would then not be covered.
+func raise(message: felt) {
+    with_attr error_message("{message}") {
+        [ap] = 0, ap++;
+        [ap] = 1, ap++;
+        dw 0x40127ffe7fff7fff;
+    }
+}
+
+raise_jmp:
```

Comment:
why do we need both raise and raise_jmp?

Reply:
a function needs to end with either a jmp, a ret or a dw. The high level `return` would check for signature consistency, the `ret` would add an uncovered line